syntax = "proto3";

package sparkplug;

message Metric {
    string name = 1;                   // The name of the metric
    int64 timestamp = 2;               // The timestamp in milliseconds since epoch

    oneof value {
        float float_value = 3;         // Float value
        int64 int_value = 4;           // Integer value
        string string_value = 5;       // String value
    }
}

message Payload {
    repeated Metric metrics = 1;       // List of metrics
}
